partner_contact_sale_info_propagation/models/res_partner.py

```diff
@@ -1,50 +1,64 @@
+                and not self.env.context.get("prevent_propagation", False)
+            )
+        )
+
     @api.multi
     def write(self, vals):
         """Propagate Salesperson and Sales Channel change in the partner to the
         child contacts."""
-        for record in self:
-            if 'user_id' in vals:
-                childs = record.mapped('child_ids').filtered(
-                    lambda r: not r.user_id or r.user_id == record.user_id)
-                if childs:
-                    childs.write({'user_id': vals['user_id']})
-            if 'team_id' in vals:
-                childs = record.mapped('child_ids').filtered(
-                    lambda r: not r.team_id or r.team_id == record.team_id)
-                if childs:
-                    childs.write({'team_id': vals['team_id']})
-        return super(ResPartner, self).write(vals)
+        if self._check_propagation_allowed():
+            for record in self:
+                if 'user_id' in vals:
+                    childs = record.mapped('child_ids').filtered(
+                        lambda r: not r.user_id or r.user_id == record.user_id)
+                    if childs:
+                        childs.write({'user_id': vals['user_id']})
+                if 'team_id' in vals:
+                    childs = record.mapped('child_ids').filtered(
+                        lambda r: not r.team_id or r.team_id == record.team_id)
+                    if childs:
+                        childs.write({'team_id': vals['team_id']})
+        return super().write(vals)
 
     @api.model
     def create(self, vals):
-        if 'parent_id' in vals:
-            if 'user_id' not in vals:
-                vals.update(user_id=self.browse(vals['parent_id']).user_id.id)
-            if 'team_id' not in vals:
-                vals.update(team_id=self.browse(vals['parent_id']).team_id.id)
-        return super(ResPartner, self).create(vals)
+        if self._check_propagation_allowed():
+            if 'parent_id' in vals:
+                if 'user_id' not in vals:
+                    vals.update(user_id=self.browse(vals['parent_id']).user_id.id)
+                if 'team_id' not in vals:
+                    vals.update(team_id=self.browse(vals['parent_id']).team_id.id)
+        return super().create(vals)
 
     @api.onchange('parent_id')
     def onchange_parent_id(self):
         """Change Salesperson or Sales Channel if the parent company changes
         and there's no Salesperson or Sales Channel defined yet"""
-        res = super(ResPartner, self).onchange_parent_id()
-        if self.parent_id and self.parent_id != self:
-            parent = self.parent_id
-            if not self.user_id:
-                res.setdefault('value', {}).update(user_id=parent.user_id)
-            if not self.team_id:
-                res.setdefault('value', {}).update(team_id=parent.team_id)
+        res = super().onchange_parent_id()
+        if self._check_propagation_allowed():
```

Comment:
Is this also needed?

Reply:
I have verified that this is not necessary.